asf_core_data/pipeline/preprocessing/feature_engineering.py

```diff
@@ -270,6 +266,14 @@ def get_heating_features(df, fine_grained_HP_types=False):
                 source_type = "electric"
                 has_hp = True
 
+            # Warm air
+            # --------------------------
+
+            elif "warm air" in heating:
+                system_type = "warm air"
+                source_type = "electric"
+                has_hp = False
+
```

Comment:
I changed this from the beginning to here to allow for all the HP search first. This way when the `MAINHEAT_DESCRIPTION` contains something like "Ground source heat pump, wairm air..." it's tagged as a ground source HP.